### PR TITLE
fix: ReportScreen is blank, indefinitely loading on web/desktop

### LINKED ISSUE #2327: Web - Chat- Loading spinner displayed on login in conversation screen
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
Last opened chat displayed

## Actual Result:
Continuous loading spinner displayed until user opens a chat

## Action Performed:
1. Launch the URL and login
2. Verify the convo screen

## Workaround:
Clicking in a chat dismisses the infinite loading icon. 

## Platform:
Where is this issue occurring?

 Web ✔️ 
 iOS
 Android
 Desktop App
 Mobile Web

**Version Number:** 1.0.18-0

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** 

image

https://user-images.githubusercontent.com/44479856/114239770-8144bb80-9954-11eb-8998-35bccbc43a73.mp4

Not sure if related to this PR https://github.com/Expensify/Expensify.cash/pull/2221 since the issue is reproducible in production. 

**Expensify/Expensify Issue URL:**

### PR DESCRIPTION
**Pull request by kidroca**
<If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit.>
@marcaaron 

### Details
Addresses the regresion found for #2221 
The details and the steps are the same as in that PR

### Fixed Issues
Fixes #2327

### Tests
On web/desktop

1. Be logged out 
2. Log in
3. Soon after loading should stop and you should see the chat view
4. There should not be a loader, loading indefinitely

Also please cover the steps in #2221 

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

File: src/pages/home/ReportScreen.js [hunk 1]
```diff
@@ -50,8 +56,8 @@ class ReportScreen extends React.Component {
      * @returns {Number}
      */
     getReportID() {
-        const params = this.props.route.params;
-        return Number.parseInt(params.reportID, 10);
+        const id = this.props.route.params.reportID || this.props.reportID;
```
Comment (kidroca) [inline src/pages/home/ReportScreen.js:59]: Currently during the first moments of this screen's life, routing would not provide `reportID` so we fall back to props.

Comment (marcaaron) [reply src/pages/home/ReportScreen.js:64]: Nice thinking! Good compromise for now for sure.

**Review by marcaaron — DISMISSED** [review]
LGTM

**Comment by kidroca** [comment]
Updated
I'm done pushing more stuff, unless you see something that needs changing here

File: src/styles/styles.js [hunk 2]
```diff
@@ -1260,7 +1260,7 @@ const styles = {
     },
 
     fullScreenLoading: {
-        backgroundColor: themeColors.modalBackdrop,
+        backgroundColor: themeColors.componentBG,
         opacity: 0.8,
```
Comment (marcaaron) [inline src/styles/styles.js:1264]: Hmm actually I think maybe we can just use a solid background or just get rid of the effect entirely on wider screens. There is still a weird flashing effect I think...

**With transparency**
https://user-images.githubusercontent.com/32969087/114217650-24b9b000-9904-11eb-97c4-85b6c9758c1a.mp4

**Without transparency**
https://user-images.githubusercontent.com/32969087/114217700-3307cc00-9904-11eb-876f-ba573622502f.mp4

**No loader**
https://user-images.githubusercontent.com/32969087/114217721-3b600700-9904-11eb-8dd2-0b1c280fe0e4.mp4


cc @shawnborton

**Comment by shawnborton** [comment]
Hmm so wait, why even have a loader at all?

**Comment by kidroca** [comment]
The issue was with the old chat being briefly visible and then a lot of heavy rendering would make the chat appear in chunks

https://user-images.githubusercontent.com/12156624/112819293-1ca07b80-908d-11eb-8164-cdd7082e3e78.mp4

The idea was to end up with something like this, but it can't be all addressed in a single PR as it's complex

https://user-images.githubusercontent.com/12156624/112987643-a32b8a80-916b-11eb-98cc-adeaaf8cfab0.mp4

We have come this far:

https://user-images.githubusercontent.com/12156624/114086691-98839a80-98bb-11eb-8768-11a96b166c55.mov

**Comment by shawnborton** [comment]
Got it. Yeah, I think fading in/out might help. Then this way if the transition is able to happen fast enough (like the internet connection is fast) then maybe we don't even get a chance to see the loader.

**Comment by jboniface** [comment]
Any chance that the update this PR is addressing led to the choppy behavior we're seeing on iOS in this video?


https://user-images.githubusercontent.com/24399930/114238440-9caec700-9952-11eb-8ea0-d867504f4caa.MP4


cc @joaniew

**Comment by kidroca** [comment]
@jboniface 
> Any chance that the update this PR is addressing led to the choppy behavior we're seeing on iOS in this video?

It doesn't seem the video includes the changes from this PR as it would display a spinner, fading out, for at least a split second

**Comment by kidroca** [comment]
@marcaaron 
I had an alternative loading handling, it makes the component a bit more complex, but will hide the loading as soons as there is data from Onyx, which would be instantly for web/desktop. You can take a look here:
https://github.com/kidroca/Expensify.cash/commit/ae376b4048c997df18781d99bff10aa81f72c4d1

**Comment by kidroca** [comment]
This indefinitely loading issue should be fixed by #2346 . unless we want to apply some small changes to the full screen loader - color, opacity, remove it, we can close this PR.

If this will be kept open I'd like to reset it to master and just apply the loader changes

**Comment by marcaaron** [comment]
@kidroca I think based on this conversation here we are expecting to see some color updates with just a softer background.

**Comment by marcaaron** [comment]
Which is pretty much what we've got and just need to fix conflicts and remove the unneeded code fixed in the other PR (now merged).

File: src/styles/styles.js (hunk 2 above)
Comment (kidroca) [reply src/styles/styles.js:1255]: Resolving this as per https://github.com/Expensify/Expensify.cash/pull/2322#issuecomment-819049488

**Comment by kidroca** [comment]
Updated. 
The PR now contains only a bg color change to the fullscreen loader

**Review by marcaaron — APPROVED** [review]

**Pull request merged**